website/docs/extensions/write/onboarding-workflow.md

```diff
@@ -0,0 +1,531 @@
+keywords: [podman desktop, extension, writing, onboarding]
+---
+
+# Onboarding
+
+A Podman Desktop extension can offer an onboarding workflow to guide users in installing and setting up all the necessary tools for the extension to work, and/or to provide explanations about the capabilities of the extension.
+
+Adding onboarding to an extension is as simple as writing JSON in the `package.json`. Podman Desktop will convert the JSON object into actual code to render all items.
+
+Onboarding consists of a title, a description, media (image), an enablement clause, and a list of steps. Before getting into the details, let's examine the JSON schema.
+
+```json
+{
+  "title": "onboarding",
+  "type": "object",
+  "properties": {
+    "title": {
+      "type": "string"
+    },
+    "description": {
+      "type": "string"
+    },
+    "media": {
+      "path": {
+        "type": "string"
+      },
+      "altText": {
+        "type": "string"
+      }
+    },
+    "enablement": {
+      "type": "string"
+    },
+    "steps": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string"
+          },
+          "title": {
+            "type": "string"
+          },
+          "description": {
+            "type": "string"
+          },
+          "media": {
+            "path": {
+              "type": "string"
+            },
+            "altText": {
+              "type": "string"
+            }
+          },
+          "command": {
+            "type": "string"
+          },
+          "completionEvents": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          },
+          "content": {
+            "type": "array",
+            "items": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "value": {
+                    "type": "string"
+                  },
+                  "highlight": {
+                    "type": "boolean"
+                  },
+                  "when": {
+                    "type": "string"
+                  }
+                },
+                "required": ["value"]
+              }
+            }
+          },
+          "when": {
+            "type": "string"
+          },
+          "component": {
+            "type": "string",
+            "enum": ["createContainerProviderConnection", "createKubernetesProviderConnection"]
+          },
+          "state": {
+            "type": "string",
+            "enum": ["completed", "failed"]
+          }
+        },
+        "required": ["id", "title"]
+      }
+    }
+  },
+  "required": ["title", "enablement", "steps"]
+}
+```
+
+### Title, Description and Media
+
+The **title**, the **description** and the **media** are all placed in the top left of the onboarding page.
+Only the title is required. The description and the media are optional.
```

Comment:
I feel like there's some missing information where we explain why only these are mandatory, or talk about the expected flow in the steps up front (maybe not here). e.g. a paragraph that says  'you're going to define a number of steps. some will be content explaining things to the user and some will be commands to execute installation...'

Reply:
new additions based on this.

1. mandatory infos -> https://github.com/containers/podman-desktop/pull/4837/files#diff-c601b46e20c5d105647d4b1ba181908e98e1b4c99f7c52983a49bf6cdfd20900R15
2. steps -> https://github.com/containers/podman-desktop/pull/4837/files#diff-c601b46e20c5d105647d4b1ba181908e98e1b4c99f7c52983a49bf6cdfd20900R149-R150

Reply:
Thank you, that helps